autoscale_cloudroast/test_repo/autoscale/fixtures.py

```diff
@@ -25,7 +25,104 @@
+    config, in some cases.
+    """
+    url = None
+
+    # If not production or staging, always use the configured server
+    # endpoint for autoscale instead of what's in the service catalog.
+    if (service_name == "autoscale" and
+            autoscale_config.environment not in ('production', 'staging')):
+        url = "{0}/{1}".format(
+            autoscale_config.server_endpoint, autoscale_config.tenant_id)
+        print(" ------ Using non-production, non-staging otter --------")
+
+    else:
+        service = access_data.get_service(service_name)
+        if service is not None:
+            url = service.get_endpoint(region).public_url
+
+    if url is not None:
+        return client_cls(url, access_data.token.id_, 'json', 'json')
+    else:
+        print("This account does not support {0}.".format(debug_name))
+
+
+def _set_up_clients():
+    """
+    Function
+    """
+    user_config = UserConfig()
+    access_data = AuthProvider.get_access_data(endpoint_config, user_config)
+
+    if access_data is None:
+        raise Exception(
+            "Unable to authenticate against identity to get auth token and "
+            "service catalog.")
+
+    autoscale_client = _make_client(
+        access_data,
+        autoscale_config.autoscale_endpoint_name,
+        autoscale_config.region,
+        AutoscalingAPIClient,
+        "Autoscale")
+
+    server_client = _make_client(
+        access_data,
+        autoscale_config.server_endpoint_name,
+        autoscale_config.server_region_override or autoscale_config.region,
+        ServersClient,
+        "Nova Compute")
+
+    lbaas_client = _make_client(
+        access_data,
+        autoscale_config.load_balancer_endpoint_name,
+        autoscale_config.lbaas_region_override or autoscale_config.region,
+        LbaasAPIClient,
+        "Cloud Load Balancers")
+
+    rcv3_client = None
+    if _rcv3_cloud_network and _rcv3_load_balancer_pool:
+        rcv3_client = _make_client(
+            access_data,
+            autoscale_config.rcv3_endpoint_name,
+            autoscale_config.rcv3_region_override or autoscale_config.region,
+            RackConnectV3APIClient,
+            "RackConnect v3")
+
+    if not all([x is not None for x in (autoscale_client, server_client,
+                                        lbaas_client)]):
+        raise Exception(
+            "Unable to instantiate all necessary clients.")
+
+    return (autoscale_client, server_client, lbaas_client, rcv3_client)
+
+
+# Global testing state - unfortunate, but only needs to be done once and also
+# makes it easier to skip tests based on configs and clients.
+
 autoscale_config = AutoscaleConfig()
+endpoint_config = UserAuthConfig()
+
+# Get optional RCV3 values.  These might not be present in the config
+# file.
+try:
+    _rcv3_load_balancer_pool = json.loads(
+        autoscale_config.rcv3_load_balancer_pool)
+except Exception:
+    _rcv3_load_balancer_pool = None
```

Comment:
I added, but in a different place (where i'm actually checking for it), if that's ok.  :)